File: Toropova/lab1.m
[X, Fs, nbits, opts] = wavread('8000.wav');
part_num = 1
X_cut = X(part_num*Fs/100:(part_num+1)*Fs/100);
X_cut_ham = hamming(length(X_cut)).*X_cut
length(X)/Fs
y = abs(fft(X, 1024));
y = abs(fft(X_cut, 1024));
y_cut_ham = abs(fft(X_cut_ham, 1024));
I = (1:length(y)/2)/length(y);
half_y = y(1:length(y)/2);
log_y = 10*log10(half_y);
hold('on')
plot(I,log_y, 'color', 'r')

y = y_cut_ham
half_y = y(1:length(y)/2);
log_y = 10*log10(half_y);
plot(I,log_y, 'color', 'b')
hold('off')

File: Timofeev/lab1.R
require(phonTools)

s<-loadsound("a.wav")
findformants(s)

require(tuneR)
require(seewave)
require(signal)

f <- readWave("a.wav")
sprintf("length = %d ms",length(f@left)/f@samp.rate * 1000)
Fs<-s[[2]]
part_num <-1
X <- f@left
X_cut <- X[part_num*Fs/100:(part_num+1)*Fs/100];
X_cut_ham <- hamming(length(X_cut))*X_cut

y <- abs(fft(X, 1024));
y_cut <- abs(fft(X_cut, 1024));
y_cut_ham <- abs(fft(X_cut_ham, 1024));

plot(y)

n <- length(f@left)
p <- fft(f@left)
nUniquePts <- ceiling((n+1)/2)
p <- p[1:nUniquePts]
p <- abs(p)
p <- p / n 
p <- p^2

if (n %% 2 > 0){
  p[2:length(p)] <- p[2:length(p)]*2
} else {
  p[2: (length(p) -1)] <- p[2: (length(p) -1)]*2
}

freqArray <- (0:(nUniquePts-1)) * (f@samp.rate / n)

plot(freqArray/1000, 10*log10(p), type='l', col='black', xlab='Frequency (kHz)', ylab='Power (dB)')